### PR TITLE
Update sigstore/sigstore to pick up TUF refresh change

### PR DESCRIPTION
**Pull request by haydentherapper**
Needed since the sscaffolding prober depends on Cosign

Signed-off-by: Hayden Blauzvern <hblauzvern@google.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -28,7 +28,7 @@ require (
 	github.com/secure-systems-lab/go-securesystemslib v0.4.0
 	github.com/sigstore/fulcio v1.0.0
 	github.com/sigstore/rekor v1.0.1
-	github.com/sigstore/sigstore v1.5.0
+	github.com/sigstore/sigstore v1.5.1-0.20230117171511-8134411c70e1
```
Comment (cpanato) [inline go.mod:31]: let me cut a sigstore release

Comment (haydentherapper) [reply go.mod:31]: thanks!

Comment (cpanato) [reply go.mod:31]: you can use https://github.com/sigstore/sigstore/releases/tag/v1.5.1

Comment (haydentherapper) [reply go.mod:31]: you're awesome, thanks :)

**Comment by haydentherapper** [comment]
Ready for lgtm and merge

**Review by cpanato — APPROVED** [review]

**Pull request merged**